appcontainer/ThemeModel.js

```diff
@@ -32,10 +32,25 @@ export class ThemeModel extends HoistModel {
         classList.toggle('xh-dark', value);
         classList.toggle('bp3-dark', value);
         this.darkTheme = value;
-        XH.setPref('xhTheme', value ? 'dark' : 'light');
+
+    }
+
+    @action
+    setThemePref(value) {
+        if (value === 'system') {
+            this.setDarkTheme(window.matchMedia('(prefers-color-scheme: dark)').matches);
```

Comment:
FYI the `matchMedia` API seems well-established / supported as per https://caniuse.com/matchmedia, so I don't _think_ we need to defend against mobile or other wonky browsers throwing here.